### PR TITLE
chore: add nx to monorepo

### PR DESCRIPTION
**Pull request by ahkhanjani**
The first step of transitioning from Lerna to Nx. This currently doesn't have a breaking effect. To use Nx tools in vscode install "Nx Console" extension.

### CODE REVIEW

**Comment by ahkhanjani** [comment]
Seems like for some reason npm messed up the lock-files. I'll make another commit

**Comment by heath-freenome** [comment]
@ahkhanjani A previous commit is actually breaking the build. I am about to merge a PR with the fix for it. You may have to wait a few hours and do a rebase.

**Comment by ahkhanjani** [comment]
Thanks for telling. I will try to fix some issues on my side as well. Please consider reading [this document](https://nx.app/docs/add-nx-cloud-to-workspace) for the free cloud caching by Nx. This is going to be added with this PR being merged and will speed up the build process.

**Comment by heath-freenome** [comment]
@ahkhanjani did you want to join our weekly meeting on August 5th at 11am Eastern Time (see https://github.com/rjsf-team/react-jsonschema-form/issues/2417 for details)?

**Comment by ahkhanjani** [comment]
> @ahkhanjani did you want to join our weekly meeting on August 5th at 11am Eastern Time (see #2417 for details) to walk us through your PR and where you see things going?

I'm afraid my time zone doesn't let me. But I can provide a written documentation and milestone.

File: nx.json [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+          "test:watch",
+          "precommit",
+          "publish-to-npm",
+          "test-coverage",
+          "build:lib",
+          "prepublishOnly",
+          "build:dist",
+          "publish-to-gh-pages"
+        ],
+        "accessToken": "N2UzOTU5MGQtMmM1OS00MTM2LWJjM2QtZDZhM2IxOWYzM2RkfHJlYWQtd3JpdGU="
```
Comment (heath-freenome) [inline nx.json:23]: Is this a secret that needs to be passed in as a variable? I'm not sure we want to use the cloud... and if we did, we really don't want this secret sitting in a public file

**Comment by heath-freenome** [comment]
> > @ahkhanjani did you want to join our weekly meeting on August 5th at 11am Eastern Time (see #2417 for details) to walk us through your PR and where you see things going?
> 
> I'm afraid my time zone doesn't let me. But I can provide a written documentation and milestone.

That would be great!

**Comment by heath-freenome** [comment]
@ahkhanjani Looks like you have conflicts... Rebase onto the latest and when you resolve the `SubmitButton` take the one from the `rjsf-v5` branch as it is. Probably the same for the `package-lock.json` files as well.

**Comment by ahkhanjani** [comment]
I will create a new pr as the current rebase ended up in a mess.

**Pull request closed without merging**